modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/search/spi/model/query/contributor/DDMFormInstanceRecordKeywordQueryContributor.java

```diff
@@ -29,16 +39,104 @@ public void contribute(
 		String keywords, BooleanQuery booleanQuery,
 		KeywordQueryContributorHelper keywordQueryContributorHelper) {
 
+		SearchContext searchContext =
+			keywordQueryContributorHelper.getSearchContext();
+
 		queryHelper.addSearchTerm(
-			booleanQuery, keywordQueryContributorHelper.getSearchContext(),
-			Field.USER_NAME, false);
+			booleanQuery, searchContext, Field.USER_NAME, false);
 
-		queryHelper.addSearchLocalizedTerm(
-			booleanQuery, keywordQueryContributorHelper.getSearchContext(),
-			"ddmContent", false);
+		_addSearchLocalizedTerm(booleanQuery, Field.TITLE, searchContext);
+		_addSearchLocalizedTerm(booleanQuery, "ddmContent", searchContext);
 	}
 
 	@Reference
 	protected QueryHelper queryHelper;
 
+	private void _addLocalizedFields(
+		BooleanQuery booleanQuery, String fieldName,
+		SearchContext searchContext, String value) {
+
+		for (String localizedFieldName :
+				_searchLocalizationHelper.getLocalizedFieldNames(
+					new String[] {fieldName}, searchContext)) {
+
+			try {
+				booleanQuery.addTerm(localizedFieldName, value, false);
+			}
+			catch (ParseException parseException) {
+				if (_log.isDebugEnabled()) {
+					_log.debug(
+						StringBundler.concat(
+							"Unable to add search term to query field:",
+							localizedFieldName, " value:", value, " like:",
+							false),
+						parseException);
+				}
+			}
+		}
+	}
+
+	private void _addLocalizedQuery(
+		BooleanQuery booleanQuery, BooleanQuery localizedQuery,
+		SearchContext searchContext) {
+
+		BooleanClauseOccur booleanClauseOccur = BooleanClauseOccur.SHOULD;
+
+		if (searchContext.isAndSearch()) {
+			booleanClauseOccur = BooleanClauseOccur.MUST;
+		}
+
+		try {
+			booleanQuery.add(localizedQuery, booleanClauseOccur);
+		}
+		catch (ParseException parseException) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(
+					StringBundler.concat(
+						"Unable to add localized localized query ",
+						localizedQuery, " with boolean clause occur ",
+						booleanClauseOccur),
+					parseException);
+			}
+		}
+	}
+
+	private void _addSearchLocalizedTerm(
+		BooleanQuery booleanQuery, String fieldName,
+		SearchContext searchContext) {
+
+		if (Validator.isBlank(fieldName)) {
+			return;
+		}
```

Comment:
@G-Sa in which case will `fieldName` be blank?

Reply:
We don't have one currently. I'll remove this validation for now